packages/app/src/device-manager/services/device-manager.service.ts

```diff
@@ -25,6 +25,8 @@ export class DeviceManager {
   public currentDevice: Device | undefined
   public devicesMap = new Map<string, Device>()
   public currentDeviceInitializationFailed = false
+  private lastSerialNumber = ""
+  private lastProductId = ""
```

Comment:
```suggestion
  private lastDeviceId = "" // as `${lastSerialNumber}-${lastProductId}`
```

I suggest considering the use of a single field `lastDeviceId` instead of two (`lastSerialNumber` and `lastProductId`)

Why it's important in my opinion:

1. **Code Clarity:** One field with a meaningful name can be more readable than two separate variables, especially when both values are related.
2. **Easier Management:** Grouping data into a single structure facilitates its management and reduces the risk of errors when manipulating these values.
3. **Conciseness:** Using template strings enables an elegant and concise way of combining values, contributing to clearer code.

Reply:
If necessary, i would suggest to use complex type like lastIds: {productId: string, serialNumber: string}, because in code those fields are used separately. If stored as template string we would need to extract single value from string.
For me those two fields are fine :)